### PR TITLE
Improve lookup table API validation

### PR DESCRIPTION
**Pull request by millerdev**
Prevent lookup tables from getting into bad states via API updates.

https://dimagi.atlassian.net/browse/SAAS-15415

:tropical_fish: Review by commit.

### Safety story

This adds a framework for API validation and fixes a few bugs. Changes are covered by tests.

### Automated test coverage

Yes.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/fixtures/resources/v0_1.py [hunk 1]
```diff
@@ -108,12 +109,55 @@ class Meta(CustomResourceMeta):
...
+                        "properties": {
+                            "type": "array",
+                            "items": {"type": "string"},
+                        },
+                    },
+                    "additionalProperties": False,
+                    "oneOf": [
+                        {"not": {"required": ["field_name"]}},
+                        {"not": {"required": ["name"]}},
+                    ],
```
Comment (millerdev) [inline corehq/apps/fixtures/resources/v0_1.py:154]: This is technically an API change, although it worked inconsistently before, and in the case of updates "field_name" was not accepted at all.

**Review by snopoke — APPROVED** [review]

File: corehq/apps/api/tests/lookup_table_resources.py [hunk 2]
```diff
@@ -249,6 +249,23 @@ def test_update_field_name(self):
...
+        lookup_table = {
+            "fields": [{"name": "property", "field_name": "prop"}],
+            "tag": "lookup_table",
+        }
+
+        response = self._assert_auth_post_resource(
+            self.single_endpoint(self.data_type.id), json.dumps(lookup_table), method="PUT")
+        self.assertEqual(response.status_code, 400)
+        errors = json.loads(response.content.decode("utf-8"))
+        print(errors)
```
Comment (snopoke) [inline corehq/apps/api/tests/lookup_table_resources.py:262]: nit: print

File: corehq/apps/api/tests/lookup_table_resources.py [hunk 3]
```diff
@@ -371,6 +388,27 @@ def test_update_field_value(self):
...
+            "field_value": "Mass...",
+        }
+        response = self._assert_auth_post_resource(
+            self.single_endpoint(data_item.id.hex),
+            json.dumps(data_item_update),
+            method="PUT",
+        )
+        self.assertEqual(response.status_code, 400)
+        errors = json.loads(response.content.decode("utf-8"))
+        print(errors)
```
Comment (snopoke) [inline corehq/apps/api/tests/lookup_table_resources.py:405]: nit: print

**Review by snopoke — APPROVED** [review]

File: corehq/apps/api/tests/lookup_table_resources.py (hunk 2 above)
Comment (millerdev) [reply corehq/apps/api/tests/lookup_table_resources.py:262]: I left these in because they are very helpful when debugging test failures.

**Pull request merged**